qiskit_experiments/library/randomized_benchmarking/mirror_rb_experiment.py

```diff
@@ -0,0 +1,522 @@
+            backend: The backend to run the experiment on.
+            num_samples: Number of samples to generate for each
+                         sequence length.
+            seed: Optional, seed used to initialize ``numpy.random.default_rng``.
+                  when generating circuits. The ``default_rng`` will be initialized
+                  with this seed value everytime :meth:`circuits` is called.
+            full_sampling: If True all Cliffords are independently sampled for
+                           all lengths. If False for sample of lengths longer
+                           sequences are constructed by appending additional
+                           Clifford samples to shorter sequences.
+            inverting_pauli_layer: If True, a layer of Pauli gates is appended at the
+                                   end of the circuit to set all qubits to 0 (with
+                                   possibly a global phase)
+
+        Raises:
+            QiskitError: if an odd length or a negative two qubit gate density is provided
+        """
+        # All lengths must be even
+        if not all(length % 2 == 0 for length in lengths):
+            raise QiskitError("All lengths must be even")
+
+        # Two-qubit density must be non-negative
+        if two_qubit_gate_density < 0:
+            raise QiskitError("Two-qubit gate density must be non-negative")
+
+        super().__init__(
+            qubits,
+            lengths,
+            backend=backend,
+            num_samples=num_samples,
+            seed=seed,
+            full_sampling=full_sampling,
+        )
+
+        self._local_clifford = local_clifford
+        self._pauli_randomize = pauli_randomize
+        self._two_qubit_gate_density = two_qubit_gate_density
+
+        # By default, the inverting Pauli layer at the end of the circuit is not added
+        self._inverting_pauli_layer = inverting_pauli_layer
+
+        # Set analysis options
+        self.analysis = MirrorRBAnalysis()
+
+    def _sample_circuits(self, lengths, rng) -> List[QuantumCircuit]:
+        """Sample Mirror RB circuits.
+
+        Steps:
+        1. Sample length/2 layers of random Cliffords
+        2. Compute inverse of each layer in the first half of the circuit and append to circuit
+        3. Sample the random paulis and interleave them between the Clifford layers
+        4. Sample the 1-qubit local Clifford and add them to the beginning and end of the circuit
+
+        Args:
+            lengths: List of lengths to run Mirror RB
+            rng: Generator seed
+
+        Returns:
+            List of QuantumCircuits
+
+        Raises:
+            QiskitError: if backend without a coupling map is provided
+        """
+
+        # Backend must have a coupling map
+        if not self._backend or not self._backend.configuration().coupling_map:
+            raise QiskitError(
+                "Must provide a backend with a coupling map or provide "
```

Comment:
If a coupling map is not provided, you can generate all possible combination instead. Such data can be easily generated with `itertools`.

Reply:
Done